src/routes/correlate.rs

```diff
@@ -0,0 +1,174 @@
+use crate::routes::ServerError;
+use crate::DbPool;
+
+use lofi::correlate::correlate_trekkie_run;
+use tlms::locations::gps::GpsPoint;
+use tlms::management::user::AuthorizedUser;
+use tlms::telegrams::r09::R09SaveTelegram;
+use tlms::trekkie::TrekkieRun;
+
+use actix_identity::Identity;
+use actix_web::{web, HttpRequest};
+use diesel::{ExpressionMethods, QueryDsl, RunQueryDsl};
+use log::{error, warn};
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+use uuid::Uuid;
+
+/// Model to correlate runs for given user. If get_result is true, the stops.json also returned
+#[derive(Serialize, Deserialize, ToSchema, Debug)]
+pub struct CorrelatePlease {
+    /// ID of run to correlate
+    pub run_id: Uuid,
+    /// Optional value for the corr_window
+    pub corr_window: Option<i64>,
+}
+
+/// Response to explicit correlate request
+#[derive(Serialize, Deserialize, ToSchema, Debug)]
+pub struct CorrelateResponse {
+    pub success: bool,
+    pub new_raw_transmission_locations: i64,
+}
+
+/// This endpoint would correlate runs for given user id. For regular user only own runs
+/// can be correlated, for admin - any run for any user
+#[utoipa::path(
+    post,
+    path = "/run/correlate",
+    responses(
+        (status = 200, description = "Correlation Successful", body = CorrelateResponse),
+        (status = 401, description = "Unauthorized"),
+        (status = 403, description = "Forbidden"),
+        (status = 500, description = "Interal Error"),
+        (status = 501, description = "Not Implemented"),
+    ),
+)]
+pub async fn correlate_run(
+    pool: web::Data<DbPool>,
+    user: Identity,
+    _req: HttpRequest,
+    corr_request: web::Json<CorrelatePlease>,
+) -> Result<web::Json<CorrelateResponse>, ServerError> {
+    // get connection from the pool
+    let mut database_connection = match pool.get() {
+        Ok(conn) => conn,
+        Err(e) => {
+            error!("cannot get connection from connection pool {:?}", e);
+            return Err(ServerError::InternalError);
+        }
+    };
+
+    // Parse the user id
```

Comment:
https://github.com/tlm-solutions/datacare/blob/master/src/routes/auth.rs#L42 maybe smart to up stream this